pkgs/development/libraries/ffmpeg/generic.nix

```
@@ -464,9 +464,10 @@ stdenv.mkDerivation (finalAttrs: {
         hash = "sha256-sqUUSOPTPLwu2h8GbAw4SfEf+0oWioz52BcpW1n4v3Y=";
       })
     ]
-    ++ optionals (lib.versionAtLeast version "7.1") [
+    ++ optionals (lib.versionAtLeast version "7.1" && lib.versionOlder version "7.1.1") [
       ./fix-fate-ffmpeg-spec-disposition-7.1.patch
-
+    ]
+    ++ optionals (lib.versionAtLeast version "7.1.1") [
```

Any reason the version requirement was set higer than it was before?

```suggestion
    ++ optionals (lib.versionAtLeast version "7.1") [
```

---

First, I should point out that `./fix-fate-ffmpeg-spec-disposition-7.1.patch` is [included in 7.1.1](https://git.ffmpeg.org/gitweb/ffmpeg.git/patch/9fbbd924f2863312b41303a48705b252b7ee1193) but not in 7.1.

What happens is that this patch must be included in 7.1 to avoid a build error for HandBrake (which does version pinning to FFmpeg 7.1), and possibly other non-nixpkgs packages that might rely on the `ffmpeg/generic.nix` interface (as discussed in #383233).

However, this patch must be excluded from 7.1.1, because applying the same patch twice results in a build error.

---

I'm was asking about why `add-av_stream_get_first_dts-for-chromium.patch` is now only included in versions after `7.1.1` and not `7.1` anymore,

---

Oh, now I see what you mean. Exposing the private API for Chromium / Qt WebEngine is not necessary for FFmpeg 7.1. But it doesn't hurt either. I've accepted your proposed change as it makes the pull request more cautious.

---

FWIW my intention with that conditional was to have only the FFmpeg our Chromium builds actually use expose the private API to minimize the risk of other things using it unnecessarily. But it’s also not a big deal by any means; it’d apply further back as well.

---

Thanks for the explanation, after reading it i would be in favor of following @emilazy's intention on this and maybe adding a comment about it.

---

I'm not sure that I'm the right person to chime in on this. @emilazy: Feel free to leave this as resolved, or to suggest any changes that would match your intention.